programs/cleos/main.cpp

```diff
@@ -492,6 +495,16 @@ fc::variant push_transaction( signed_transaction& trx, const std::vector<public_
                std::cerr << "New RPC compute_transaction may not be supported. Submit to a different node." << std::endl;
                throw;
             }
+         } else if( tx_read ) {
+            EOSC_ASSERT( !tx_retry_lib, "ERROR: --retry-irreversible can not be used with --read" );
+            EOSC_ASSERT( !tx_retry_num_blocks, "ERROR: --retry-num-blocks can not be used with --read" );
+            try {
+               auto args = fc::mutable_variant_object ("transaction", packed_transaction(trx,compression));
+               return call( send_read_only_txn_func, args );
+            } catch( chain::missing_chain_api_plugin_exception& ) {
+               std::cerr << "New RPC readonly_transaction may not be supported. Submit to a different node." << std::endl;
+               throw;
+            }
```

Comment:
Seems like this could be combined with the above `if` and remove some duplicate code.

Reply:
Thanks.